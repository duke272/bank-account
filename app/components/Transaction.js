import React from 'react';
import PropTypes from 'prop-types';
import { Divider, ListItem, ListItemText } from '@material-ui/core';
import EuroIcon from '@material-ui/icons/Euro';

const Transaction = ({ transaction }) => (
  <>
    <ListItem>
      <ListItemText
        primary={
          <>
            <EuroIcon fontSize="inherit" />
            {transaction.formattedAmount}
          </>
        }
        secondary={
          <>
            <span title="omschrijving">{transaction.description}</span>
            <br />
            <span title="naam">{transaction.name}</span>
            <br />
            {transaction.iban && <span title="iban">{transaction.iban}</span>}
          </>
        }
      />
    </ListItem>
    <Divider />
  </>
);

Transaction.propTypes = {
  transaction: PropTypes.shape({
    formattedAmount: PropTypes.string,
    description: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    iban: PropTypes.string,
  }),
};

export default Transaction;
