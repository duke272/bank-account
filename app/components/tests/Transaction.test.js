import React from 'react';
import renderer from 'react-test-renderer';
import Transaction from '../Transaction';

describe('<Transaction />', () => {
  it('should render with an Iban number', () => {
    const transWithIban = {
      formattedAmount: '+44,33',
      description: 'description of transaction',
      name: 'Henk de Vries',
      iban: 'NL22RABO3154100547',
    };
    const tree = renderer
      .create(<Transaction transaction={transWithIban} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without an Iban number', () => {
    const transWithoutIban = {
      formattedAmount: '+44,33',
      description: 'description of transaction',
      name: 'Henk de Vries',
    };
    const tree = renderer
      .create(<Transaction transaction={transWithoutIban} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
