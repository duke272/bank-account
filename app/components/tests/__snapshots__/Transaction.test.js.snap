// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transaction /> should render with an Iban number 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57z"
          />
        </svg>
        +44,33
      </span>
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        <span
          title="omschrijving"
        >
          description of transaction
        </span>
        <br />
        <span
          title="naam"
        >
          Henk de Vries
        </span>
        <br />
        <span
          title="iban"
        >
          NL22RABO3154100547
        </span>
      </p>
    </div>
  </li>,
  <hr
    className="MuiDivider-root"
  />,
]
`;

exports[`<Transaction /> should render without an Iban number 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57z"
          />
        </svg>
        +44,33
      </span>
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        <span
          title="omschrijving"
        >
          description of transaction
        </span>
        <br />
        <span
          title="naam"
        >
          Henk de Vries
        </span>
        <br />
      </p>
    </div>
  </li>,
  <hr
    className="MuiDivider-root"
  />,
]
`;
